Mathlib/MeasureTheory/Measure/Comap.lean

```diff
@@ -129,6 +130,20 @@ theorem comap_preimage (f : α → β) (μ : Measure β) (hf : Injective f) (hf'
   · simp [comap, hf]
   · simp [comap, hf]
 
+@[simp]
+lemma comap_id (μ : Measure β) : comap (fun x ↦ x) μ = μ := by
+  ext s hs
+  rw [comap_apply, image_id']
+  · exact injective_id
+  all_goals simp [*]
+
+lemma comap_comap (hf : Injective f) (hf' : ∀ s, MeasurableSet s → MeasurableSet (f '' s))
```

Comment:
Can you case-split on `Injective f`? if `f` is not injective, then nor is `g ∘ f` and so both sides are 0, right?

Reply:
Well spotted. I was looking for a generalisation along those lines